### PR TITLE
[Merged by Bors] - feat: add smul_prod lemmas

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
From #6718.

---

- [x] depends on: #8936

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by eric-wieser — COMMENTED** [review]
Do we already have the finset version? If so, can it be golfed using the multiset result?

File: Mathlib/Algebra/BigOperators/Basic.lean [hunk 1]
```diff
@@ -1994,6 +1994,13 @@ theorem prod_int_mod (s : Finset α) (n : ℤ) (f : α → ℤ) :
   (Multiset.prod_int_mod _ _).trans <| by rw [Finset.prod, Multiset.map_map]; rfl
 #align finset.prod_int_mod Finset.prod_int_mod
 
+@[to_additive (attr := simp)]
+theorem smul_prod
+    [CommMonoid α] [Monoid β] [MulAction β α] [IsScalarTower β α α] [SMulCommClass β α α]
+    (s : Finset α) (b : β) :
+    ∏ x in s, b • x = b ^ card s • ∏ x in s, x := by
```
Comment (eric-wieser) [inline Mathlib/Algebra/BigOperators/Basic.lean:2001]: Thinking again, `smul_prod` doesn't make sense here because we have `smul_sum : r • ∑ x in s, f x = ∑ x in s, r • f x` which is the other way around. So either we should reverse the direction of the lemmas, or restore your original names.

**Review by eric-wieser — COMMENTED** [review]
It would be nice to have the `∏ x in s, x • c` lemmas too, if you'll willing to let the scope expand.

**Comment by Ruben-VandeVelde** [comment]
> It would be nice to have the `∏ x in s, x • c` lemmas too, if you'll willing to let the scope expand.

That one is there, no?

File: Mathlib/Algebra/BigOperators/Basic.lean [hunk 2]
```diff
@@ -1994,6 +1994,13 @@ theorem prod_int_mod (s : Finset α) (n : ℤ) (f : α → ℤ) :
   (Multiset.prod_int_mod _ _).trans <| by rw [Finset.prod, Multiset.map_map]; rfl
 #align finset.prod_int_mod Finset.prod_int_mod
 
+@[to_additive (attr := simp)]
+theorem smul_prod
+    [CommMonoid α] [Monoid β] [MulAction β α] [IsScalarTower β α α] [SMulCommClass β α α]
+    (s : Finset α) (b : β) :
+    b ^ card s • ∏ x in s, x = ∏ x in s, b • x := by
```
Comment (urkud) [inline Mathlib/Algebra/BigOperators/Basic.lean:2001]: I'm not sure that RHS is simpler than LHS. In many cases, I would move in the inverse direction.

Comment (eric-wieser) [reply Mathlib/Algebra/BigOperators/Basic.lean:2001]: This order is to match an existing order elsewhere; so I'd be inclined to just remove `simp`. Agreed about `f` and the other variant.

**Comment by riccardobrasca** [comment]
Can you summarize in the PR description the difference between `Finset.smul_prod` and `Finset.smul_prod'` (and the variants for lists and multisets). Thanks!

**Comment by grunweg** [comment]
Coming here from PR triage: it seems like you have one outstanding review ask, and you have a merge conflict now. Can you fix both, please? Thanks!
#6718 is huge (but would also be great to have in mathlib) - it would be great for at least make steady progress on pieces like this.

**Comment by Ruben-VandeVelde** [comment]
> Can you summarize in the PR description the difference between `Finset.smul_prod` and `Finset.smul_prod'` (and the variants for lists and multisets). Thanks!

I summarized in the module documentation instead

**Comment by riccardobrasca** [comment]
Thanks!

bors merge
